### ISSUE TITLE
An empty MiningSchema is a deviation from the PMML standard

### ISSUE THREAD

**Issue by afard** [jpmml/jpmml-sklearn#173]
First, I am a big fan of the JPMML project, and I'm excited to observe its growth during the last years.

Recently, I tried jpmml-sklearn to convert an XGB model to its PMML format. Following, I add the Python script. I noticed that some of the tree models have an empty MiningSchema. An empty MiningSchema is a deviation from the PMML standard.
Here is the format of MiningSchema specified by the standard (https://dmg.org/pmml/v4-4-1/MiningSchema.html):

<xs:element name="MiningSchema">
  <xs:complexType>
    <xs:sequence>
      <xs:element ref="Extension" minOccurs="0" maxOccurs="unbounded"/>
      <xs:element maxOccurs="unbounded" ref="MiningField"/>
    </xs:sequence>
  </xs:complexType>
</xs:element>

Based on XML Schema guideline, the default value of minOccurs is 1 (https://www.w3.org/TR/xmlschema-0/#:~:text=The%20default%20value%20for%20both,not%20occur%20more%20than%20once.). Therefor, a MiningSchema must have at least one MiningField. We may file an issue on the Github page of JPMML.

It is correct that the tree models with empty MiningSchema have only a root node (no use of any input field), but they could still have defined MiningFields. In other words, a model can have defined inputs that are not used inside the model for prediction. To best of my understanding, a model as a black box in the PMML standard cannot be without any inputs; regardless of the fact how those inputs are utilized.

# The Python script
from sklearn import datasets
import pandas as pd
import xgboost as xgb

iris = datasets.load_iris()
print(iris.target_names)
print(iris.feature_names)
print(iris.data[0:5])
print(iris.target)

data=pd.DataFrame({
    'sepal length':iris.data[:,0],
    'sepal width':iris.data[:,1],
    'petal length':iris.data[:,2],
    'petal width':iris.data[:,3],
    'species':iris.target
})
data.head()
from sklearn.model_selection import train_test_split

X=data[['sepal length', 'sepal width', 'petal length', 'petal width']]  # Features
y=data['species']  # Labels
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3) # 70% training and 30% test
clf=xgb.XGBClassifier()
clf.fit(X_train,y_train)
y_pred=clf.predict(X_test)
from sklearn import metrics
print("Accuracy:",metrics.accuracy_score(y_test, y_pred))

from sklearn2pmml import sklearn2pmml, make_pmml_pipeline
pmml_obj = make_pmml_pipeline(clf)  
sklearn2pmml(pmml_obj,"xgb_iris.xml")
print("Model exported as PMML successfully.Filename: xgb_iris.xml " )
